x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java

```diff
@@ -548,6 +549,28 @@ private void handleTokenResponse(HttpResponse httpResponse, ActionListener<Tuple
         }
     }
 
+    private static String truncateJWT(JWT jwt) {
```

Comment:
why do we truncate the ID Token string?

Reply:
I think "leak" is a strong word here. The ID Token contains user information that the OP is willing to share with the RP and that the user has consented to. Furthermore, this is only logged on `TRACE`, which requires an explicit configuration action and, lastly, we do log the claims from the ID Token in at least one other place here.  

I see no reason to truncate the ID Token :)